packages/headless/src/features/commerce/facets/facet-set/facet-set-slice.ts

```diff
@@ -35,14 +49,39 @@ export const commerceFacetSetReducer = createReducer(
         const {facetId, selection} = action.payload;
         const facetRequest = state[facetId]?.request;
 
-        if (!facetRequest) {
+        if (!facetRequest || facetRequest.type !== 'regular') {
           return;
         }
 
         facetRequest.preventAutoSelect = true;
 
-        const existingValue = facetRequest.values.find(
-          (req) => req.value === selection.value
+        const existingValue = (
+          facetRequest.values as WritableDraft<FacetValueRequest>[]
+        ).find((req) => req.value === selection.value);
+        if (!existingValue) {
+          insertNewValue(facetRequest, selection);
+          return;
+        }
+        const isSelected = existingValue.state === 'selected';
+        existingValue.state = isSelected ? 'idle' : 'selected';
+      })
+      .addCase(toggleSelectNumericFacetValue, (state, action) => {
+        const {facetId, selection} = action.payload;
+        const facetRequest = state[facetId]?.request;
+
+        if (!facetRequest || facetRequest.type !== 'numericalRange') {
+          return;
+        }
+
+        facetRequest.preventAutoSelect = true;
+
+        const existingValue = (
+          facetRequest.values as WritableDraft<NumericRangeRequest>[]
+        ).find(
+          (req) =>
+            req.start === selection.start &&
+            req.end === selection.end &&
+            req.endInclusive === selection.endInclusive
```

Comment:
In the findRange method in range-facet-reducers.ts, we're not considering the endInclusive value when finding the range value. I added this check because 0-3 end inclusive is completely different from 0-3 not end inclusive.